src/components/Filters/index.tsx

```diff
@@ -1,20 +1,61 @@
 import Button, { ButtonSize, ButtonVariants } from 'components/shared/Buttons';
 import InputField, { InputVariants } from 'components/shared/InputFields';
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import { FiltersType } from 'pages/ProductsPage/productspage';
+import { useGetCategoriesQuery } from 'services/blackMarketApi';
 
-const Filters: React.FC = () => {
-  const [condition, setCondition] = useState('');
-  const [category, setCategory] = useState('');
+type Props = {
+  filters: FiltersType;
+  setFilters: (filters: FiltersType) => void;
+};
+
+const conditionMapping: { [key: string]: string } = {
+  New: 'N',
+  Used: 'U',
+  Refurbished: 'A',
+};
+
+type Category = {
+  id: number;
+  name: string;
+};
+
+type CategoryData = {
+  results?: Category[];
+};
+
+const Filters: React.FC<Props> = ({ filters, setFilters }) => {
+  const [condition, setCondition] = useState<keyof typeof conditionMapping>('');
+  const [categories, setCategories] = useState(['All']);
   const [minPrice, setMinPrice] = useState('');
   const [maxPrice, setMaxPrice] = useState('');
 
-  const submitFilters = () => {
-    const filters = {
-      condition,
-      category,
-      price: { min: minPrice, max: maxPrice },
+  const { data: categoryData = { results: [] } as CategoryData } = useGetCategoriesQuery({
+    page: 1,
+    page_size: 10,
+  });
+
+  useEffect(() => {
+    const newFilters: FiltersType = {
+      ...filters,
+      categories: categories[0] !== 'All' ? categories : [],
+      state: condition ? conditionMapping[condition] : '',
+      unit_price_min: minPrice ? parseFloat(minPrice) : null,
+      unit_price_max: maxPrice ? parseFloat(maxPrice) : null,
     };
-    console.log(filters);
+
+    setFilters(newFilters);
+  }, [condition, filters, categories, minPrice, maxPrice, setFilters]);
+
+  const resetFilters = () => {
+    setCondition('');
+    setCategories(['All']);
+    setMinPrice('');
+    setMaxPrice('');
+  };
+
+  const isFilterApplied = () => {
+    return condition !== '' || categories[0] !== 'All' || minPrice !== '' || maxPrice !== '';
   };
```

Comment:
It might be a good idea to move all of the filtering logic into a `useFilters` hook, instead of having it inside a component. It might be an overkill for the scope of the blackmarket but it's usually how it should be done

Reply:
You could also have an object like `filtersInitialState` so you can use it to initiate the states, in the reset filters function and in the isFilterApplied function